### PR TITLE
refactor(proxy): introduce Context variants

### PR DESCRIPTION
**Pull request by geigerzaehler**
We make the `Context` type into an enum so that the API is forced to handle the case where the keystore has not been unsealed yet and consequently the coco state and peer are not available.

Currently, the API is always intialized with an unsealed context. Making dynamic changes will be introduced in a follow-up.

### CODE REVIEW

**Review by xla — APPROVED** [review]
Very welcome and anticipated change.

🔓 📇 🏜 💱

File: proxy/api/src/http/session.rs [hunk 1]
```diff
@@ -48,25 +48,30 @@ fn update_settings_filter(
...
 
     /// Fetch the [`session::Session`].
     pub async fn get(ctx: context::Context) -> Result<impl Reply, Rejection> {
-        let sess = session::current(ctx.state.clone(), &ctx.store).await?;
-
-        Ok(reply::json(&sess))
+        match ctx {
+            context::Context::Unsealed(unsealed) => {
+                let sess = session::current(unsealed.state.clone(), &unsealed.store).await?;
+
```
Comment (NunoAlexandre) [inline proxy/api/src/http/session.rs:65]:
```suggestion
```

**Review by NunoAlexandre — APPROVED** [review]
Solid :ok_hand:

**Review by MeBrei — APPROVED** [review]
⭐

**Review by xla — DISMISSED** [review]
😏 ◽️ 😄 💮

**Review by NunoAlexandre — DISMISSED** [review]
:surfing_man:

File: proxy/api/src/http/session.rs [hunk 2]
```diff
@@ -48,25 +48,29 @@ fn update_settings_filter(
...
 
         Ok(reply::with_status(reply(), StatusCode::NO_CONTENT))
     }
 
     /// Fetch the [`session::Session`].
     pub async fn get(ctx: context::Context) -> Result<impl Reply, Rejection> {
-        let sess = session::current(ctx.state.clone(), &ctx.store).await?;
-
-        Ok(reply::json(&sess))
+        match ctx {
```
Comment (FintanH) [inline proxy/api/src/http/session.rs:62]: Why isn't this taking `Unsealed` and using the guard?

**Review by FintanH — COMMENTED** [review]
Just one question :)

File: proxy/api/src/http/session.rs (hunk 2 above)
Comment (geigerzaehler) [reply proxy/api/src/http/session.rs:62]: Excellent catch! The code here is indeed a copy `with_context_unsealed`. I’ve pushed a fix

**Review by xla — APPROVED** [review]
💰 💒 👹 🐲

**Review by FintanH — APPROVED** [review]
:metal:

**Pull request merged**